### ISSUE TITLE
[cmd/telemetrygen] Use standard SDK configuration

### ISSUE THREAD

**Issue by axw** [open-telemetry/opentelemetry-collector-contrib#35736]
### Component(s)

cmd/telemetrygen

### Is your feature request related to a problem? Please describe.

Right now telemetrygen has various non-standard flags for configuring
 - OTLP exporter: `--otlp-endpoint`, `--otlp-header`, `--mtls`, etc.
 - SDK processors: `--batch`
 - Resource attributes: `--otlp-attributes`, `--service`

Having non-standard configuration adds some cognitive overhead for users of the tool, as they must learn about the tool's specific configuration.

### Describe the solution you'd like

I would like telemetrygen to support standard OTel SDK configuration: https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/config. That way end users have a single standard configuration format for the things described above.

When adding support for standard config, deprecate existing telemetrygen flags (except those that are specific to telemetrygen logic), and remove after a grace period.

### Describe alternatives you've considered

Use https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/exporters/autoexport, which supports a standard set of env vars for configuring the SDK: https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/. This approach is being replaced by the config file approach mentioned above.

_No response_

**Comment by mx-psi** [comment]
If we go down this route, should we support the [OTel SDK configuration file](https://github.com/open-telemetry/opentelemetry-configuration) instead? This is meant to be the replacement for environment variables

**Comment by axw** [comment]
@mx-psi sounds sensible. I'll rename the issue to be less about env vars, and more about "standard SDK configuration". What I really intended was for telemetrygen to follow an SDK standard.

I had a read of the config repo & OTEP, and couldn't find a clear answer: will there be no standard env vars in the long run? I certainly see the value in having a config file format, and understand that env vars are not expressive enough for all configuration, but I think there's also value in having standard default config that supports the current env vars for simple things such as the OTLP endpoint.

Has there been any discussion of having a default, built-in config in SDKs, e.g. https://github.com/open-telemetry/opentelemetry-configuration/blob/main/examples/sdk-migration-config.yaml?

Any pointers to relevant docs/specs/discussions would be appreciated.

**Comment by mx-psi** [comment]
@codeboten can probably answer these questions better than I can, but I believe this is meant to be (once stable) *the* recommended way of configuring the SDK. [OTEP 225](https://github.com/open-telemetry/oteps/blob/main/text/0225-configuration.md) has probably useful context

**Comment by axw** [comment]
@dmathieu pointed out to me that the [environment variable specification](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/) is stable, so removing/breaking them would require a major version bump.

**Comment by mx-psi** [comment]
@axw Yes, the env var spec would still be supported, what I meant with the recommended way was the way that is used in all examples and documentation for example.

**Comment by Erog38** [comment]
Is this still preferred? I tend away from config files for a CLI tool like this, as it makes pulling a recently used command from history more difficult as you may not have the original config you had used. 

If this was just an addition to support a config, that could make sense.

**Comment by axw** [comment]
@Erog38 I do too, and from my understanding there should be an embedded configuration file that supports injecting configuration with standard environment variables.

I'm not pushing for this change at the moment, as we settled on using an opentelemetry-collector for load generation: https://github.com/elastic/opentelemetry-collector-components/tree/main/loadgen

I still think the issue makes sense though -- telemetrygen is still be useful for ad-hoc use cases.
